controlpanel/frontend/views/app.py

```diff
@@ -295,6 +295,21 @@ def form_invalid(self, form):
         raise Exception(form.errors)
 
 
+class EnableBedrockApp(OIDCLoginRequiredMixin, PermissionRequiredMixin, UpdateView):
```

Comment:
This is largely a copy and paste of the `EnableBedrockUser` view right? So maybe to keep DRY you could create a `BedrockAccessMixin` to have the shared methods, so that we can reduce the views themselves to:
```
class EnableBedrockApp(BedrockAccessMixin, UpdateView):
    model = App


class EnableBedrockApp(BedrockAccessMixin, UpdateView):
    model = User
```

To do this you would need to [add `get_absolute_url` methods](https://docs.djangoproject.com/en/5.0/ref/models/instances/#get-absolute-url) to the `User` and `App` models, and then those methods will be used by the `UpdateView` [when `get_success_url` is called](https://ccbv.co.uk/projects/Django/5.0/django.views.generic.edit/UpdateView/#get_success_url). Let me know if you have any questions or want to pair on this.